tools/const_finder/lib/const_finder.dart

```diff
@@ -33,8 +33,14 @@ class _ConstVisitor extends RecursiveVisitor<void> {
   final List<Map<String, dynamic>> nonConstantLocations;
 
   bool _matches(Class node) {
-    return node.enclosingLibrary.importUri.toString() == classLibraryUri &&
-      node.name == className;
+    if (node == null) {
+      return false;
+    }
+    final bool exactMatch = node.name == className
+        && node.enclosingLibrary.importUri.toString() == classLibraryUri;
+    return exactMatch
+        || _matches(node.superclass)
+        || node.implementedTypes.any((Supertype implementedType) => _matches(implementedType.classNode));
```

Comment:
This might be a quite expensive to traverse supers for each instance constant and constructor invocation.
Consider creating ClassHierarchy object (see pkg/kernel/lib/class_hierarchy.dart) and using ClassHierarchy.isSubtypeOf for this check, or at least cache results in a Map<Class, bool>.

Also, consider using
```dart
  node.supers.any((Supertype sup) => _matches(sup.classNode))
```
which includes superclass, implementedTypes and mixedInType which was missing.

Reply:
Done